core/shared/src/main/scala/hedgehog/core/PropertyT.scala

```diff
@@ -168,14 +168,12 @@ trait PropertyTReporting {
 
   def report(config: PropertyConfig, size0: Option[Size], seed0: Seed, p: PropertyT[Result]): Report = {
     // Increase the size proportionally to the number of tests to ensure better coverage of the desired range
-    val sizeInc = Size(Math.max(1, Size.max / config.testLimit.value))
-    // Start the size at whatever remainder we have to ensure we run with "max" at least once
-    val sizeInit = Size(Size.max % Math.min(config.testLimit.value, Size.max)).incBy(sizeInc)
+    val sizeInc = Size(Size.max / config.testLimit.value)
+    // The increment value is also the starting value because size doesn't start at 0.
+    val sizeInit = sizeInc
```

Comment:
Double-check my logic (I checked a little in the REPL):

* `Math.min(config.testLimit.value, Size.max)` means 100+
* `Size.max % Math.min(config.testLimit.value, Size.max)` means `100 % 100+`, which is always 0
* (the `/ config.testLimit.value` above implies it's, at least, not 0 - could still be negative though...)
* `Size(0).incBy(sizeInc)` (before my value-correcting change here) meant simply `sizeInc`